neo/Ledger/Blockchain.cs

```diff
@@ -604,6 +604,8 @@ private void Persist(Block block)
                     snapshot.HeaderHashIndex.GetAndChange().Hash = block.Hash;
                     snapshot.HeaderHashIndex.GetAndChange().Index = block.Index;
                 }
+                foreach (IPersistencePlugin plugin in Plugin.PersistencePlugins)
+                    plugin.OnPersist(snapshot);
                 snapshot.Commit();
```

Comment:
We want to allow the plugin to alter the snapshot? maybe is better to commit first?

Reply:
Shargons broda, we need this before the commit because we want to track state differences (Added, Changed of Deleted).

It will be an useful Plugin not only for storage purposes.
It is a good catch that Erik Yodas had.

Reply:
Is this call synchronous? A persistence plugin should be able to crash / halt the whole node to ensure integrity of its data in relation to the node.